openshift/assisted-service
# MGMT-9021: Set feature usage for ignition config overrides

[MGMT-9021](https://issues.redhat.com/browse/MGMT-9021)
Sets the feature usage for ignition config overrides when
it's specified by the user.



## List all the issues related to this PR


## What environments does this code impact?


## How was this code tested?




Creating an infra-env
1. Deploy service with changes
2. Curl the api to create a cluster
3. Curl the api to create an infra-env with an ignition config override 
4. Get the cluster to see that the feature usage flag was added

Updating an infra-env
1. Deploy service with changes
2. Curl the api to create a cluster
3. Curl the api to create an infra-env without an ignition config override 
4. Get the cluster to see that the feature usage flag was not added
5. Patch the infra-env with an ignition config override
4. Get the cluster to see that the feature usage flag was added

## Assignees



/cc @carbonin 
/cc @omertuc 

## Checklist


## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md
